src/ProcessSlave.php

```diff
@@ -291,37 +291,49 @@ protected function sendCurrentFiles()
      */
     public function run()
     {
-        $this->loop = Factory::create();
+        if (!$this->loop) {
+            $this->loop = Factory::create();
+        }
 
         $this->errorLogger = BufferingLogger::create();
         ErrorHandler::register(new ErrorHandler($this->errorLogger));
 
         $connector = new UnixConnector($this->loop);
         $unixSocket = $this->getControllerSocketPath(false);
 
-        $connector->connect($unixSocket)->done(
-            function ($controller) {
-                $this->controller = $controller;
+        try {
+            $connector->connect($unixSocket)->done(
+                function ($controller) {
+                    $this->controller = $controller;
 
-                $pcntl = new PCNTL($this->loop);
-                $pcntl->on(SIGTERM, [$this, 'shutdown']);
-                $pcntl->on(SIGINT, [$this, 'shutdown']);
-                register_shutdown_function([$this, 'shutdown']);
+                    $pcntl = new PCNTL($this->loop);
```

Comment:
Why is this line changed? Whitespace?

Reply:
This was all nested in a try block, so it's correct.